graviti/dataset/dataset.py

```diff
@@ -41,55 +40,95 @@ def __init__(
         *,
         status: Status,
         alias: str,
-        is_public: bool,
-        config_name: str,
     ) -> None:
-        pass
+        self._access_key = access_key
+        self._url = url
+        self._dataset_id = dataset_id
+        self._name = name
+        self._status = status
+        self._alias = alias
 
     @property
-    def url(self) -> str:
-        """Return the url of the graviti website.
+    def access_key(self) -> str:
+        """Return the access key of the user.
 
-        Return:
-            The url of the graviti website.
+        Returns:
+            The access key of the user.
 
         """
+        return self._access_key
 
     @property
-    def access_key(self) -> str:
-        """Return the access key of the user.
+    def url(self) -> str:
+        """Return the url of the graviti website.
 
-        Return:
-            The access key of the user.
+        Returns:
+            The url of the graviti website.
 
         """
+        return self._url
 
     @property
     def dataset_id(self) -> str:
         """Return the ID of the dataset.
 
-        Return:
+        Returns:
             The ID of the dataset.
 
         """
+        return self._dataset_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the dataset.
+
+        Returns:
+            The name of the dataset.
+
+        """
+        return self._name
 
     @property
     def status(self) -> Status:
         """Return the status of the dataset.
 
-        Return:
+        Returns:
             The status of the dataset.
 
         """
+        return self._status
+
+    @property
+    def alias(self) -> str:
+        """Return the alias of the dataset.
+
+        Returns:
+            The alias of the dataset.
+
+        """
+        return self._alias
+
+    @property
+    def is_public(self) -> bool:
+        """Return whether the dataset is public.
+
+        Returns:
+            Whether the dataset is public.
+
+        """
+        info = get_dataset(url=self.url, access_key=self.access_key, dataset_id=self.dataset_id)
+        return info["isPublic"]  # type: ignore[no-any-return]
 
     @property
     def config_name(self) -> str:
         """Return the config name of the dataset.
 
-        Return:
+        Returns:
```

Comment:
I think config_name should be redesigned, we can remove this property in this PR.